src/tmpfiles/tmpfiles.c

```diff
@@ -835,35 +842,58 @@ static int fd_set_perms(Item *i, int fd, const struct stat *st) {
         return label_fix(path, 0);
 }
 
-static int path_set_perms(Item *i, const char *path) {
-        _cleanup_close_ int fd = -1;
-        struct stat st;
+static int path_open_parent_safe(const char *path) {
+        _cleanup_free_ char *dn = NULL;
+        int fd;
+
+        if (path_equal(path, "/") || !path_is_normalized(path))
+                return -EINVAL;
+
+        dn = dirname_malloc(path);
+        if (!dn)
+                return -ENOMEM;
+
+        fd = chase_symlinks(dn, NULL, CHASE_OPEN|CHASE_SAFE, NULL);
+        if (fd == -EPERM)
+                return log_error_errno(fd, "Unsafe symlinks encountered in %s, refusing.", path);
+        if (fd < 0)
+                return log_error_errno(fd, "Failed to validate path %s: %m", path);
+
+        return fd;
```

Comment:
path_open_parent_safe() now logs some failures but not others. We generally try to make sure that functions either log about all errors or about none. Otherwise it's too likely that some errors might be logged about twice and possibly others non at all.

if figure it would be better if this function would log in all cases, hence, please fix the early two checks to log properly.

Also see CODING_STYLE about this for an explanation